mwdb/web/src/components/RecentView/RecentView.js

```diff
@@ -159,7 +230,11 @@ export default function RecentView(props) {
                             type={props.type}
                             query={submittedQuery}
                             canAddQuickQuery={canAddQuickQuery}
-                            submitQuery={(q) => setCurrentQuery(q)}
+                            submitQuery={(query) =>
+                                setCurrentQuery({
+                                    query: query,
```

Comment:
That one can be shortened
```suggestion
                                    query,
```